airflow/api_connexion/endpoints/dag_endpoint.py

```diff
@@ -34,14 +34,14 @@
 
 
 @security.requires_access([(permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG)])
-@provide_session
-def get_dag(dag_id, session):
+def get_dag(dag_id):
     """Get basic information about a DAG."""
-    dag = session.query(DagModel).filter(DagModel.dag_id == dag_id).one_or_none()
-
+    try:
+        dag: DAG = current_app.dag_bag.get_dag(dag_id)
+    except SerializedDagNotFound:
+        raise NotFound("DAG not found", detail=f"The DAG with dag_id: {dag_id} was not found")
     if dag is None:
         raise NotFound("DAG not found", detail=f"The DAG with dag_id: {dag_id} was not found")
-
     return dag_schema.dump(dag)
```

Comment:
This serializer accepts DagModel. See: https://github.com/apache/airflow/blob/fcae7a2d85b5bd1423ea904b6ef7441e33aeb988/airflow/api_connexion/schemas/dag_schema.py#L47

Reply:
If I'm not mistaken the `get_dag` method indeed returns a DAG object, so instead of updating the typing, should I query it from the database after validating its existence on the DagBag?